spacy_llm/backends/integration/dolly.py

```diff
@@ -78,9 +90,6 @@ def backend_dolly_hf(
     if not config or len(config) == 0:
         config = _compile_default_config()
 
-    llm_pipeline = transformers.pipeline(model=model, **config)
-
-    def query(prompts: Iterable[str]) -> Iterable[str]:
-        return [llm_pipeline(pr)[0]["generated_text"] for pr in prompts]
-
-    return query
+    return Backend["transformers.Pipeline", str, str](
```

Comment:
Fair point regarding `transformers.Pipeline`. It's not necessary to maintain to this information for the purpose of type-checking.

We don't want to enforce `_PromptType` and `_ResponseType` to be `str` though, they can't be anything.

> Then we make a subclass for each backend to handle its init.

This would be IMO unnecessary overhead, each of these classes will do exactly the same thing.

Reply:
Looking at how to do this without generics.

Reply:
As a middle ground, we could define a:

```python
from .base import Backend
from ...compat import transformers

DollyBackend = Backend["transformers.Pipeline", str, str]
```

Then the logic wouldn't be so cluttered. Although between that and a `Protocol`, the debate seems fair.

Reply:
The core issues here are:
1. we need to use global `Callable`s to avoid the multiprocessing issue
2. we need type information to be available for our introspective type checks

`Protocol` wouldn't help us with #2. We'd need to write a separate `Protocol` per backend, which amounts to Kabir's original suggestion.

Reply:
I rewrote this to work without `Generic`s. There's only a little bit of weird stuff happening in the type checks, which I think is an ok trade-off to have. 

Let me know what you think @bdura @kabirkhan.

Reply:
`# type: ignore[arg-type]` is too bad...

Reply:
Yeah, I don't know why `mypy` complains about a type mismatch otherwise - it should recognize that these are `TypeVar`s, I think.